pkg/kubectl/stop.go

```diff
@@ -88,12 +90,29 @@ func (reaper *ReplicationControllerReaper) Stop(namespace, name string, timeout
 	if err != nil {
 		return "", err
 	}
+	ctrl, err := rc.Get(name)
+	if err != nil {
+		return "", err
+	}
 	if timeout == 0 {
-		rc, err := rc.Get(name)
-		if err != nil {
-			return "", err
+		timeout = Timeout + time.Duration(10*ctrl.Spec.Replicas)*time.Second
+	}
+
+	// Completely back off till the user has deleted all overlapping controllers. The rc manager
+	// handles overlapping controllers by sorting on creation timestamp and only syncing the
+	// first one. Unless the controller being deleted is also the controller being synced by the
+	// rc manager, stopping it will not converge.
+	overlappingCtrl, err := getReplicationControllersForLabels(rc, labels.Set(ctrl.Spec.Selector))
+	if err != nil {
+		return "", err
+	}
+	if len(overlappingCtrl) > 1 {
+		names := []string{}
+		for _, c := range overlappingCtrl {
+			names = append(names, c.Name)
 		}
-		timeout = Timeout + time.Duration(10*rc.Spec.Replicas)*time.Second
+		return "", fmt.Errorf(
+			"Overlapping controllers detected %v. Please delete all the others with --cascade=false.", strings.Join(names, ","))
```

Comment:
If we can detect this situation, why not just do the right thing?
* If the rc they're trying to delete is not the active one, automatically assume --cascade=false.
* If it is the active one, everything should just work-- replicas should go to zero. It'll go up again as soon as we actually delete the RC, but that should be after the loop is satisfied?